monai/transforms/spatial/array.py

```diff
@@ -277,16 +277,17 @@ def __call__(  # type: ignore
             align_corners=align_corners,
             dtype=dtype,
         )
-        if isinstance(img, MetaTensor):
-            img.affine = dst_affine
-            if isinstance(img_dst, MetaTensor):
-                original_fname = img.meta.get(Key.FILENAME_OR_OBJ, "resample_to_match_source")
-                img.meta = deepcopy(img_dst.meta)
-                img.meta[Key.FILENAME_OR_OBJ] = original_fname  # keep the original name, the others are overwritten
+        if not self.lazy_evaluation:
```

Comment:
I'm not confident on this part. Hi @wyli  @KumoLiu , do you have any opinions?

Reply:
thanks, I add some logic here to partly copy the metadata https://github.com/Project-MONAI/MONAI/pull/6080/commits/a05d8231f5bef7f22cdb56dcfc7d601fcc16f9d0, please feel free to revise...